src/app/loadout-drawer/loadout-utils.ts

```diff
@@ -90,7 +96,8 @@ export function createSocketOverridesFromEquipped(item: DimItem) {
           // Only save them if they're valid plug options though, otherwise
           // we'd save the empty stasis sockets that Void 3.0 spawns with
           plugFitsIntoSocket(socket, socket.plugged.plugDef.hash) &&
-          socket.plugged.plugDef.hash !== socket.emptyPlugItemHash
+          socket.plugged.plugDef.hash !== socket.emptyPlugItemHash &&
+          (category.category.hash !== SocketCategoryHashes.Fragments || activeFragments-- > 0)
```

Comment:
Let me know if the short circuit + postdecrement is too clever.

Reply:
even if it's the right amount of clever it's extremely missable and would benefit from a comment